src/libs/filterArrayByMatch.ts

```diff
@@ -0,0 +1,326 @@
+    threshold?: Ranking;
+};
+
+type KeyOption<T> = KeyAttributesOptions<T> | ValueGetterKey<T> | string;
+
+type Options<T = unknown> = {
+    keys?: ReadonlyArray<KeyOption<T>>;
+    threshold?: Ranking;
+    strict?: boolean;
+};
+type IndexableByString = Record<string, unknown>;
+
+/**
+ * Generates an acronym for a string.
+ * @param string the string for which to produce the acronym
+ * @returns the acronym
+ */
+function getAcronym(string: string): string {
+    let acronym = '';
+    const wordsInString = string.split(' ');
+    wordsInString.forEach((wordInString) => {
+        const splitByHyphenWords = wordInString.split('-');
+        splitByHyphenWords.forEach((splitByHyphenWord) => {
+            acronym += splitByHyphenWord.substring(0, 1);
+        });
+    });
+    return acronym;
+}
+
+/**
+ * Given path: "foo.bar.baz" and item: {foo: {bar: {baz: 'buzz'}}} -> 'buzz'
+ * @param path a dot-separated set of keys
+ * @param item the item to get the value from
+ */
+function getNestedValues<T>(path: string, item: T): string[] {
+    const keys = path.split('.');
+    let values: Array<T | IndexableByString | string> = [item];
+
+    for (const nestedKey of keys) {
+        let nestedValues: Array<T | IndexableByString | string> = [];
+
+        for (const nestedItem of values) {
+            if (nestedItem != null) {
+                if (Object.hasOwnProperty.call(nestedItem, nestedKey)) {
+                    const nestedValue = (nestedItem as IndexableByString)[nestedKey];
+                    if (nestedValue != null) {
+                        nestedValues.push(nestedValue as IndexableByString | string);
+                    }
+                } else if (nestedKey === '*') {
+                    nestedValues = nestedValues.concat(nestedItem);
+                }
+            }
+        }
+
+        values = nestedValues;
+    }
+
+    if (Array.isArray(values[0])) {
+        const result: string[] = [];
+        return result.concat(...(values as string[]));
+    }
+
+    return values as string[];
+}
+
+/**
+ * Gets value for key in item at arbitrarily nested keypath
+ * @param item - the item
+ * @param key - the potentially nested keypath or property callback
+ * @returns an array containing the value(s) at the nested keypath
+ */
+function getItemValues<T>(item: T, key: KeyOption<T>): string[] {
+    let keyCopy = key;
+    if (typeof keyCopy === 'object') {
+        keyCopy = keyCopy.key as string;
+    }
+    // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+    let value: string | string[] | null | unknown;
+    if (typeof keyCopy === 'function') {
+        value = keyCopy(item);
+    } else if (item == null) {
+        value = null;
+    } else if (Object.hasOwnProperty.call(item, keyCopy)) {
+        value = (item as IndexableByString)[keyCopy];
+    } else if (keyCopy.includes('.')) {
+        return getNestedValues<T>(keyCopy, item);
+    } else {
+        value = null;
```

Comment:
probably you can do return []; here, instead of assign null and then check for null and return